### PR TITLE
[crashtracker] Chain signal handlers 

### PR DESCRIPTION
**Pull request by danielsn**
# What does this PR do?

Reworks the signal handler chaining code to properly handle cases where the signal was manually raised.

# Motivation

The example in `crashtracker.c` would fail to call the underlying handler in the case of a `raise(SEGV)` call, and then would continue past what ought to be a core-dump.  Now we don't do that anymore.

# Additional Notes

The POSIX signal infra is 🤯 

# How to test the change?

`crashtracking.c` now works the way its expected to.

## For Reviewers
- [ ] If this PR touches code that signs or publishes builds or packages, or handles credentials of any kind, I've requested a review from `@DataDog/security-design-and-guidance`.
- [ ] This PR doesn't touch any of that.

### CODE REVIEW

**Review by paullegranddc — APPROVED** [review]

**Comment by paullegranddc** [comment]
Could you edit the CODEOWNERS file in a separate PR to put the crashtracker crate under the responsability of Datadog/libdatadog-profiling?

File: crashtracker/src/crash_handler.rs
Comment (ivoanjo) [inline crashtracker/src/crash_handler.rs:215]: Minor: Comment cut off abruptly?

File: crashtracker/src/crash_handler.rs
Comment (ivoanjo) [inline crashtracker/src/crash_handler.rs:296]: I guess this is somewhat harmless, but wasn't the existing code already handling this correctly? E.g. the first crash would grab the receiver and any other crash would return with the "No crashtracking receiver" option.

**Review by ivoanjo — APPROVED** [review]
This looks reasonable. Two notes:

1. May be worth getting a review from the native profiler wizards (e.g. @sanchda / @nsavoire / @r1viollet )
2. I know testing this is a pain but... have you considered perhaps modifying the crash tracker example to have a few more situations?

**Review by sanchda — APPROVED** [review]
LGTM!

**Pull request merged**
